core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -563,12 +563,8 @@ private object RTS {
                   case IO.Tags.Uninterruptible =>
                     val io = curIo.asInstanceOf[IO.Uninterruptible[E, Any]]
 
-                    // FIXME: Not safe because of potential error in computing `v`
-                    curIo = for {
-                      _ <- enterUninterruptible
-                      v <- io.io
-                      _ <- exitUninterruptible
-                    } yield v
+                    // FIXME: Not safe because of potential error in computing `io.io`
+                    curIo = enterUninterruptible *> io.io <* exitUninterruptible
```

Comment:
Perhaps
```scala
curIo = enterUninterruptible *> io.io.ensuring(exitUninterruptible)
```
as suggested [here](https://github.com/scalaz/scalaz-zio/issues/68), makes the final `map` disappear

Reply:
Done. Thanks! @regiskuckaertz